src/agent/src/image_rpc.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2019 alibaba
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::fs;
+use std::process::{ExitStatus, Command};
+use std::sync::Arc;
+
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use protocols::image;
+use tokio::sync::Mutex;
+use ttrpc::{self, error::get_rpc_status as ttrpc_error};
+
+use crate::sandbox::Sandbox;
+//use crate::trace_rpc_call;
+use crate::rpc::CONTAINER_BASE;
+
+const SKOPEO_PATH: &str = "/usr/bin/skopeo";
+const UMOCI_PATH: &str = "/usr/local/bin/umoci";
+
+// Convenience macro to obtain the scope logger
+macro_rules! sl {
+    () => {
+        slog_scope::logger()
+    };
+}
+
+pub struct ImageService {
+    sandbox: Arc<Mutex<Sandbox>>,
+}
+
+impl ImageService {
+    pub fn new(sandbox: Arc<Mutex<Sandbox>>) -> Self {
+        Self { sandbox }
+    }
+
+    fn pull_image_from_registry(image: &str, cid: &str, source_creds: &Option<&str>) -> Result<()> {
+        let source_image = format!("{}{}", "docker://", image);
+
+        let manifest_path = format!("/tmp/{}/image_manifest", cid);
+        let target_path_manifest = format!("dir://{}", manifest_path);
+
+        // Define the target transport and path for the OCI image, without signature
+        let oci_path = format!("/tmp/{}/image_oci:latest", cid);
+        let target_path_oci = format!("oci://{}", oci_path);
+
+        fs::create_dir_all(&manifest_path)?;
+        fs::create_dir_all(&oci_path)?;
+
+        info!(sl!(), "Attempting to pull image {}...", &source_image);
+
+        let mut pull_command = Command::new(SKOPEO_PATH);
+        pull_command
+            .arg("--insecure-policy")
```

Comment:
This might be ok for our next demo, but we should think about a proper policy to be used if planning to go with skopeo as the final solution for pulling images.

Reply:
I feel that this will be re-implemented through `ImageManager` in the future.

Reply:
I think it's fine for now, as @bpradipt said.